### PR TITLE
add simple support pedal lines

### LINKED ISSUE #967: Implement pedal@form="altpedstar" / pedal bounces not rendered in MIDI
In the following notation example there are implicit pedal ups before the second, third and fourth pedal downs:

screen shot 2018-12-04 at 3 57 48 pm

in other words, the "Ped" symbol is saying to put the pedal down.  If the pedal is already down, then it is also implying that the pedal must be raised before pushing it down again.  The current MIDI rendering does not currently account for these implied pedal-ups.  It would be useful if the MIDI rendering from verovio added the implicit pedal-ups.

Test MEI data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2018-12-04T15:45:20" version="2.0.0-dev-a03f221">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv xml:id="mdiv-0000000373266819">
                <score xml:id="score-0000001449229880">
                    <scoreDef xml:id="scoredef-0000000521262779" midi.bpm="120">
                        <staffGrp xml:id="staffgrp-0000001566073580">
                            <staffDef xml:id="staffdef-0000001415898864" clef.shape="G" clef.line="2" meter.count="4" meter.unit="4" n="1" lines="5">
                                <label xml:id="label-0000001623559751" />
                            </staffDef>
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-L1F1">
                        <measure xml:id="measure-L4" n="1">
                            <staff xml:id="staff-L4F1N1" n="1">
                                <layer xml:id="layer-L4F1N1" n="1">
                                    <chord xml:id="chord-L6F1" dur="4">
                                        <note xml:id="note-L6F1S1" oct="4" pname="c" accid.ges="n" />
                                        <note xml:id="note-L6F1S2" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L6F1S3" oct="4" pname="g" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L8F1" dur="4">
                                        <note xml:id="note-L8F1S1" oct="4" pname="d" accid.ges="n" />
                                        <note xml:id="note-L8F1S2" oct="4" pname="f" accid.ges="n" />
                                        <note xml:id="note-L8F1S3" oct="4" pname="a" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L10F1" dur="4">
                                        <note xml:id="note-L10F1S1" oct="4" pname="c" accid.ges="n" />
                                        <note xml:id="note-L10F1S2" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L10F1S3" oct="4" pname="g" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L12F1" dur="4">
                                        <note xml:id="note-L12F1S1" oct="4" pname="d" accid.ges="n" />
                                        <note xml:id="note-L12F1S2" oct="4" pname="f" accid.ges="n" />
                                        <note xml:id="note-L12F1S3" oct="4" pname="a" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <pedal xml:id="pedal-0000001918090860" staff="1" tstamp="1.000000" dir="down" vgrp="200" />
                            <pedal xml:id="pedal-0000001111975696" staff="1" tstamp="2.000000" dir="down" vgrp="200" />
                            <pedal xml:id="pedal-0000001792288206" staff="1" tstamp="3.000000" dir="down" vgrp="200" />
                            <pedal xml:id="pedal-0000001854520701" staff="1" tstamp="4.000000" dir="down" vgrp="200" />
                        </measure>
                        <measure xml:id="measure-L13" right="end" n="2">
                            <staff xml:id="staff-L13F1N1" n="1">
                                <layer xml:id="layer-L13F1N1" n="1">
                                    <chord xml:id="chord-L14F1" dur="1">
                                        <note xml:id="note-L14F1S1" oct="4" pname="c" accid.ges="n" />
                                        <note xml:id="note-L14F1S2" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L14F1S3" oct="4" pname="g" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <pedal xml:id="pedal-0000001307739072" staff="1" tstamp="5.000000" dir="up" vgrp="200" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

Rendering of data into MIDI with this command:

```bash
verovio -atmidi test.mei
```

Resulting MIDI file (binasc representation to view the individual messages):

```
"MThd"			; MIDI header chunk marker
4'6			; bytes to follow in header chunk
2'1			; file format: Type-1 (multitrack)
2'2			; number of tracks
2'120			; ticks per quarter note

;;; TRACK 0 ----------------------------------
"MTrk"			; MIDI track chunk marker
4'11			; bytes to follow in track chunk
v0	ff 51 v3 t120	; tempo
v0	ff 2f v0	; end-of-track

;;; TRACK 1 ----------------------------------
"MTrk"			; MIDI track chunk marker
4'145			; bytes to follow in track chunk
v0	b0 '64 '127	; controller
v0	90 '60 '64	; note-on C4
v0	90 '64 '64	; note-on E4
v0	90 '67 '64	; note-on G4
v120	b0 '64 '127	; pedal on
v0	90 '67 '0	; note-off G4
v0	90 '60 '0	; note-off C4
v0	90 '64 '0	; note-off E4
v0	90 '65 '64	; note-on F4
v0	90 '69 '64	; note-on A4
v0	90 '62 '64	; note-on D4
v120	b0 '64 '127	; pedal on
v0	90 '65 '0	; note-off F4
v0	90 '62 '0	; note-off D4
v0	90 '69 '0	; note-off A4
v0	90 '67 '64	; note-on G4
v0	90 '60 '64	; note-on C4
v0	90 '64 '64	; note-on E4
v120	b0 '64 '127	; pedal on
v0	90 '67 '0	; note-off G4
v0	90 '60 '0	; note-off C4
v0	90 '64 '0	; note-off E4
v0	90 '65 '64	; note-on F4
v0	90 '69 '64	; note-on A4
v0	90 '62 '64	; note-on D4
v120	90 '65 '0	; note-off F4
v0	90 '62 '0	; note-off D4
v0	90 '69 '0	; note-off A4
v0	90 '64 '64	; note-on E4
v0	90 '67 '64	; note-on G4
v0	90 '60 '64	; note-on C4
v480	b0 '64 '0	; pedal off
v0	90 '60 '0	; note-off C4
v0	90 '67 '0	; note-off G4
v0	90 '64 '0	; note-off E4
v0	ff 2f v0	; end-of-track
```

There are four pedal downs and one pedal up, following the explicit pedal floating-elements in the MEI data.

The reason for this request is that this is an existing syntax for pedaling in piano music which is reasonably common, such as in this edition of Hummel's prelude in D-flat major, op. 67/15:

screen shot 2018-12-04 at 4 07 44 pm

---------------------

Somewhat related to this, `pedal@dir="bounce"` does not place any pedal messages into the MIDI rendering.  They should instead insert both a pedal up and then a pedal down (at the same timestamp).

Test MEI data with bounced pedals:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2018-12-04T16:12:48" version="2.0.0-dev-a03f221">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv xml:id="mdiv-0000002021852513">
                <score xml:id="score-0000000502270845">
                    <scoreDef xml:id="scoredef-0000001412136073" midi.bpm="120">
                        <staffGrp xml:id="staffgrp-0000000292397043">
                            <staffDef xml:id="staffdef-0000000692409210" clef.shape="G" clef.line="2" meter.count="4" meter.unit="4" n="1" lines="5">
                                <label xml:id="label-0000001021885283" />
                            </staffDef>
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-L1F1">
                        <measure xml:id="measure-L4" n="1">
                            <staff xml:id="staff-L4F1N1" n="1">
                                <layer xml:id="layer-L4F1N1" n="1">
                                    <chord xml:id="chord-L6F1" dur="4">
                                        <note xml:id="note-L6F1S1" oct="4" pname="c" accid.ges="n" />
                                        <note xml:id="note-L6F1S2" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L6F1S3" oct="4" pname="g" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L8F1" dur="4">
                                        <note xml:id="note-L8F1S1" oct="4" pname="d" accid.ges="n" />
                                        <note xml:id="note-L8F1S2" oct="4" pname="f" accid.ges="n" />
                                        <note xml:id="note-L8F1S3" oct="4" pname="a" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L10F1" dur="4">
                                        <note xml:id="note-L10F1S1" oct="4" pname="c" accid.ges="n" />
                                        <note xml:id="note-L10F1S2" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L10F1S3" oct="4" pname="g" accid.ges="n" />
                                    </chord>
                                    <chord xml:id="chord-L12F1" dur="4">
                                        <note xml:id="note-L12F1S1" oct="4" pname="d" accid.ges="n" />
                                        <note xml:id="note-L12F1S2" oct="4" pname="f" accid.ges="n" />
                                        <note xml:id="note-L12F1S3" oct="4" pname="a" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <pedal xml:id="pedal-0000001786990701" staff="1" tstamp="1.000000" dir="down" vgrp="200" />
                            <pedal xml:id="pedal-0000001747700930" staff="1" tstamp="2.000000" dir="bounce" vgrp="200" />
                            <pedal xml:id="pedal-0000001495696456" staff="1" tstamp="3.000000" dir="bounce" vgrp="200" />
                            <pedal xml:id="pedal-0000000547989941" staff="1" tstamp="4.000000" dir="bounce" vgrp="200" />
                        </measure>
                        <measure xml:id="measure-L13" right="end" n="2">
                            <staff xml:id="staff-L13F1N1" n="1">
                                <layer xml:id="layer-L13F1N1" n="1">
                                    <chord xml:id="chord-L14F1" dur="1">
                                        <note xml:id="note-L14F1S1" oct="4" pname="c" accid.ges="n" />
                                        <note xml:id="note-L14F1S2" oct="4" pname="e" accid.ges="n" />
                                        <note xml:id="note-L14F1S3" oct="4" pname="g" accid.ges="n" />
                                    </chord>
                                </layer>
                            </staff>
                            <pedal xml:id="pedal-0000001805647366" staff="1" tstamp="5.000000" dir="up" vgrp="200" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

Conversion to MIDI:

```bash
verovio -atmidi test2.mei
```

MIDI rendering only has a pedal on for the first pedal down, and then a pedal off for the pedal up at the end of the music:

```
"MThd"			; MIDI header chunk marker
4'6			; bytes to follow in header chunk
2'1			; file format: Type-1 (multitrack)
2'2			; number of tracks
2'120			; ticks per quarter note

;;; TRACK 0 ----------------------------------
"MTrk"			; MIDI track chunk marker
4'11			; bytes to follow in track chunk
v0	ff 51 v3 t120	; tempo
v0	ff 2f v0	; end-of-track

;;; TRACK 1 ----------------------------------
"MTrk"			; MIDI track chunk marker
4'133			; bytes to follow in track chunk
v0	b0 '64 '127	; pedal on
v0	90 '60 '64	; note-on C4
v0	90 '64 '64	; note-on E4
v0	90 '67 '64	; note-on G4
v120	90 '60 '0	; note-off C4
v0	90 '67 '0	; note-off G4
v0	90 '64 '0	; note-off E4
v0	90 '69 '64	; note-on A4
v0	90 '62 '64	; note-on D4
v0	90 '65 '64	; note-on F4
v120	90 '69 '0	; note-off A4
v0	90 '62 '0	; note-off D4
v0	90 '65 '0	; note-off F4
v0	90 '67 '64	; note-on G4
v0	90 '64 '64	; note-on E4
v0	90 '60 '64	; note-on C4
v120	90 '67 '0	; note-off G4
v0	90 '60 '0	; note-off C4
v0	90 '64 '0	; note-off E4
v0	90 '62 '64	; note-on D4
v0	90 '69 '64	; note-on A4
v0	90 '65 '64	; note-on F4
v120	90 '69 '0	; note-off A4
v0	90 '62 '0	; note-off D4
v0	90 '65 '0	; note-off F4
v0	90 '64 '64	; note-on E4
v0	90 '67 '64	; note-on G4
v0	90 '60 '64	; note-on C4
v480	b0 '64 '0	; pedal off
v0	90 '67 '0	; note-off G4
v0	90 '64 '0	; note-off E4
v0	90 '60 '0	; note-off C4
v0	ff 2f v0	; end-of-track
```

### LINKED ISSUE #1425: Render pedal brackets
Current rendering of this:

```
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
  <meiHead>
    <fileDesc>
      <titleStmt>
        <title/>
      </titleStmt>
      <pubStmt/>
    </fileDesc>
  </meiHead>
  <music>
    <body>
      <mdiv>
        <score>
          <scoreDef>
            <staffGrp>
              <staffDef n="1" lines="5" ppq="1" clef.shape="G" clef.line="2"
                key.sig="0" meter.count="4" meter.unit="4">
                <label>Piano</label>
              </staffDef>
            </staffGrp>
          </scoreDef>
          <section>
            <measure right="end" n="1">
              <staff n="1">
                <layer n="1">
                  <note dur.ppq="4" dur="1" oct="5" pname="c" stem.dir="down"/>
                </layer>
              </staff>
              <pedal staff="1" tstamp="1" tstamp2="5" dir="down" place="below"
                form="line"/>
            </measure>
          </section>
        </score>
      </mdiv>
    </body>
  </music>
</mei>
```

Looks like this:

grafik

A pedal bracket should be added.

### PR DESCRIPTION
**Pull request by rettinghaus**
This PR adds simple rendition for pedal lines. For now, only `@dir="down"` and `@dir="up"` is taken into account for lines. In order to support `@dir="bounce"`, some things has to be reconsidered. 
Adapts the MusicXML import according. 
In addition, now `@dir="bounce"` is taken into account for MIDI output.

closes #967
closes #1425

### CODE REVIEW

File: include/vrv/pedal.h [hunk 1]
```diff
@@ -64,14 +73,11 @@ class Pedal : public ControlElement,
...
-protected:
-    //
 private:
-    //
-public:
-    //
-private:
-    //
+    /**
+     * Flag indicating whether or not the element was generated
```
Comment (lpugin) [inline include/vrv/pedal.h:78]: The comment does not make sense, does it?

File: src/view_control.cpp [hunk 2]
```diff
@@ -1857,7 +1928,7 @@ void View::DrawPedal(DeviceContext *dc, Pedal *pedal, Measure *measure, System *
 
     int code = SMUFL_E655_keyboardPedalUp;
     std::wstring str;
-    if (pedal->GetDir() == pedalLog_DIR_bounce) {
+    if (pedal->GetDir() == pedalLog_DIR_bounce && pedal->GetForm() != pedalVis_FORM_altpedstar) {
```
Comment (lpugin) [inline src/view_control.cpp:1931]: Add additional ()

File: src/view_control.cpp [hunk 3]
```diff
@@ -1877,16 +1948,21 @@ void View::DrawPedal(DeviceContext *dc, Pedal *pedal, Measure *measure, System *
...
-            continue;
-        }
-        // Basic method that use bounding box
-        int y = pedal->GetDrawingY();
+    dc->StartGraphic(pedal, "", pedal->GetUuid());
 
-        dc->SetFont(m_doc->GetDrawingSmuflFont((*staffIter)->m_drawingStaffSize, false));
-        DrawSmuflString(dc, x, y, str, alignment, (*staffIter)->m_drawingStaffSize);
-        dc->ResetFont();
+    // Don't draw a symbol, if it's a line
```
Comment (lpugin) [inline src/view_control.cpp:1953]: Confusing, say // Draw a symbol if it is not a line
But there is something not very clear here: `str` is modified above but then not used if is it a line. I think the logic can be improved.

**Comment by rettinghaus** [comment]
Example:
pedal-003

```
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
	<meiHead>
		<fileDesc>
			<titleStmt>
				<title>Piano pedal as line</title>
			</titleStmt>
			<pubStmt>
				<date>2020-05-12</date>
			</pubStmt>
			<notesStmt>
				<annot>Pedal markings may be rendered as lines.</annot>
			</notesStmt>
		</fileDesc>
		<encodingDesc>
			<appInfo>
				<application version="2.7.0" label="2">
					<name>Verovio</name>
				</application>
			</appInfo>
		</encodingDesc>
	</meiHead>
	<music>
		<body>
			<mdiv>
				<score>
					<scoreDef>
						<staffGrp>
							<staffDef clef.shape="F" clef.line="4" meter.count="4" meter.unit="4" meter.sym="common" n="1" lines="5" />
						</staffGrp>
					</scoreDef>
					<section>
						<measure>
							<staff n="1">
								<layer>
									<note dur="4" oct="3" pname="b" />
									<note dur="4" oct="3" pname="f" />
									<note dur="4" oct="3" pname="g" />
									<note dur="4" oct="3" pname="d" />
								</layer>
							</staff>
							<pedal form="line" staff="1" tstamp="1" dir="down" />
							<pedal form="line" staff="1" tstamp="5" dir="up" />
						</measure>
						<measure>
							<staff n="1">
								<layer>
									<note dur="4" oct="3" pname="b" />
									<note dur="4" oct="3" pname="f" />
									<note dur="4" oct="3" pname="g" />
									<note dur="4" oct="3" pname="d" />
								</layer>
							</staff>
							<pedal form="line" staff="1" tstamp="1" dir="down" />
							<pedal form="line" staff="1" tstamp="2.5" dir="up" />
							<pedal form="line" staff="1" tstamp="3" dir="down" />
							<pedal form="line" staff="1" tstamp="4.5" dir="up" />
						</measure>
					</section>
				</score>
			</mdiv>
		</body>
	</music>
</mei>
```

**Pull request merged**